python/tank/deploy/tank_commands/unregister_folders.py

```diff
@@ -0,0 +1,311 @@
+                log.info("> tank unregister_folders /path/to/folder_a /path/to/folder_b")
+                log.info("")
+                
+            else:
+                paths = args
+                log.info("Will process the following folders:")
+                for p in paths:
+                    log.info(" - %s" % p)
+                log.info("")
+                self._unregister_paths(paths, log, prompt=True)
+        
+        
+    def run_noninteractive(self, log, parameters):
+        """
+        API accessor
+        
+        :param log: Std logging object
+        :param parameters: Std tank command parameters dict
+        :returns: List of dictionaries to represents the items that were unregistered.
+                  Each dictionary has keys path and sg_id.
+                  Note that the shotgun ids returned will refer to retired objects in 
+                  Shotgun rather than live ones.
+        """
+        # validate params and seed default values
+        computed_params = self._validate_parameters(parameters) 
+        
+        # must have a path or an entity
+        if computed_params["path"] is None and computed_params["entity"] is None:
+            raise TankError("Must either specify a path or an entity!")
+        # ... but not both
+        if computed_params["path"] and computed_params["entity"]:
+            raise TankError("Cannot specify both a path and an entity!")
+            
+        if computed_params["path"]:
+            path = computed_params["path"]
+            return self._unregister_paths([path], log, prompt=False)
+        
+        if computed_params["entity"]:
+            entity = computed_params["entity"]
+            if "id" not in entity:
+                raise TankError("Entity dictionary does not contain an id key!")
+            if "type" not in entity:
+                raise TankError("Entity dictionary does not contain a type key!")
+            return self._unregister_entity(entity, log, prompt=False)
+            
+
+    def _unregister_paths(self, paths, log, prompt):
+        """
+        Unregisters a path.
+        
+        :returns: List of dictionaries to represents the items that were unregistered.
+                  Each dictionary has keys path and sg_id.
+                  Note that the shotgun ids returned will refer to retired objects in 
+                  Shotgun rather than live ones.
+        """
+        log.debug("Will unregister folders for the following root paths:")
+        for p in paths:
+            log.debug(p)
+        
+        # first of all, make sure we are up to date.
+        pc = path_cache.PathCache(self.tk)
+        try:
+            pc.synchronize(log)
+
+            fs_location_ids = set()
+            for p in paths:
+                sg_id = pc.get_shotgun_id_from_path(p)
+                if sg_id is None:
+                    log.warning("Path '%s' is not registered in Shotgun - ignoring." % p)
```

Comment:
In one of your example, you have
----------------------------------------------------------------------
Command: Unregister folders
----------------------------------------------------------------------

Will process the following folders:
 - /mnt/projects/climp/sequences/aaa/aaa_00010
 - /mnt/projects/climp/sequences/aaa
 - /mnt/projects/climp/sequences/aaa
 - /foo/bar

WARNING: Path '/foo/bar' is not registered in Shotgun - ignoring.

The following folders will be unregistered:
 - /mnt/projects/climp/sequences/aaa
 - /mnt/projects/climp/sequences/aaa/aaa_00010
 - /mnt/projects/climp/sequences/aaa/aaa_00010/Anm
 - /mnt/projects/climp/sequences/aaa/aaa_00010/Comp
 - /mnt/projects/climp/sequences/aaa/aaa_00010/FX
 - /mnt/projects/climp/sequences/aaa/aaa_00010/Lay
 - /mnt/projects/climp/sequences/aaa/aaa_00010/Light
 - /mnt/projects/climp/sequences/aaa/aaa_00020
 - /mnt/projects/climp/sequences/aaa/aaa_00020/Anm
 - /mnt/projects/climp/sequences/aaa/aaa_00020/Comp
 - /mnt/projects/climp/sequences/aaa/aaa_00020/FX
 - /mnt/projects/climp/sequences/aaa/aaa_00020/Lay
 - /mnt/projects/climp/sequences/aaa/aaa_00020/Light
 - /mnt/projects/climp/sequences/aaa/aaa_00030
 - /mnt/projects/climp/sequences/aaa/aaa_00030/Anm
 - /mnt/projects/climp/sequences/aaa/aaa_00030/Comp
 - /mnt/projects/climp/sequences/aaa/aaa_00030/FX
 - /mnt/projects/climp/sequences/aaa/aaa_00030/Lay
 - /mnt/projects/climp/sequences/aaa/aaa_00030/Light
 - /mnt/projects/climp/sequences/aaa/aaa_00010
 - /mnt/projects/climp/sequences/aaa/aaa_00010/Anm
 - /mnt/projects/climp/sequences/aaa/aaa_00010/Comp
 - /mnt/projects/climp/sequences/aaa/aaa_00010/FX
 - /mnt/projects/climp/sequences/aaa/aaa_00010/Lay
 - /mnt/projects/climp/sequences/aaa/aaa_00010/Light

I think that from a readability standpoint, it might be better to display the paths that are going to be ignored (the warnings) last. The reasoning is that when looking at text inside a scrolling console, our eyes generally scroll from bottom to top since it's easier to see the end of the output than the beginning. This warning list is going to generally be shorter than the list of unregistered folders and giving them a better exposure by putting them at the end would make the output better in my opinion.

What do you think @manneohrstrom @kporangehat ?